File: sample-chat-obj-c/sample-chat/Components/ChatScreen/Views/ChatCollectionView/ChatIncomingCell/ChatIncomingCell.m
//
//  ChatIncomingCell.m
//  samplechat
//
//  Created by Injoit on 2/25/19.
//  Copyright © 2019 Quickblox. All rights reserved.
//

#import "ChatIncomingCell.h"
#import "UIView+Chat.h"
#import "CALayer+Chat.h"

@implementation ChatIncomingCell

- (void)awakeFromNib {
    [super awakeFromNib];
    
    self.previewContainer.backgroundColor = UIColor.whiteColor;
    UIColor *shadowColor = [UIColor colorWithRed:0.85f green:0.9f blue:1.0f alpha:1.0f];
    [self.layer applyShadowWithColor:shadowColor
                               alpha:1.0f
                                forX:0.0f
                                forY:3.0f
                                blur:48.0f
                              spread:0.0f
                                path:nil];
}

+ (ChatCellLayoutModel)layoutModel {
    
    ChatCellLayoutModel defaultLayoutModel = [super layoutModel];
    defaultLayoutModel.containerInsets = UIEdgeInsetsMake(0, 24, 12, 16);
    defaultLayoutModel.avatarSize = CGSizeMake(40, 40);
    return defaultLayoutModel;
}

- (void)layoutSubviews {
    CGFloat cornerRadius = 20.0f;
    CAShapeLayer *layer = CAShapeLayer.new;
    layer.frame = self.previewContainer.layer.bounds;
    UIRectCorner roundingCorners = UIRectCornerTopLeft | UIRectCornerTopRight | UIRectCornerBottomRight;
    UIBezierPath *bPath = [UIBezierPath bezierPathWithRoundedRect:self.previewContainer.bounds
                                                byRoundingCorners:roundingCorners
                                                      cornerRadii:CGSizeMake(cornerRadius, cornerRadius)];
    layer.path = bPath.CGPath;
    self.previewContainer.layer.mask = layer;
}

@end

File: sample-chat-swift/sample-chat-swift/Components/ChatScreen/Chat/CollectionView/CollectionCells/ChatIncomingCell/ChatIncomingCell.swift
//
//  ChatIncomingCell.swift
//  sample-chat-swift
//
//  Created by Injoit on 1/28/19.
//  Copyright © 2019 Quickblox. All rights reserved.
//

import UIKit

/**
 *  Chat message cell typically used for opponent's messages.
 */
class ChatIncomingCell: ChatCell {
    
    override func awakeFromNib() {
        super.awakeFromNib()
        
        previewContainer.backgroundColor = .white
        layer.applyShadow(color: #colorLiteral(red: 0.8452011943, green: 0.8963350058, blue: 1, alpha: 1), alpha: 1.0, y: 3.0, blur: 48.0)
    }
    
    override class func layoutModel() -> ChatCellLayoutModel {
        var defaultLayoutModel = super.layoutModel()
        let containerInsets = UIEdgeInsets(top: 0.0, left: 24.0, bottom: 12.0, right: 16.0)
        defaultLayoutModel.containerInsets = containerInsets
        defaultLayoutModel.avatarSize = CGSize(width: 40.0, height: 40.0)
        
        return defaultLayoutModel
    }
    
    override func layoutSubviews() {
        let roundingCorners:UIRectCorner = [.topLeft, .topRight, .bottomRight]
        let layer = CAShapeLayer()
        layer.frame = previewContainer.layer.bounds
        let bPath = UIBezierPath(roundedRect: previewContainer.bounds,
                                 byRoundingCorners: roundingCorners,
                                 cornerRadii: CGSize(width: 20, height: 20))
        layer.path = bPath.cgPath
        previewContainer.layer.mask = layer
    }
}
